src/edu/stanford/nlp/sempre/interactive/JsonFn.java

```diff
@@ -280,12 +284,17 @@ public Derivation createDerivation() {
   }
 
   // takes a token and check if it can be a path
-  static class JsonValueStream extends SingleDerivationStream {
+  static class JsonValueStream extends MultipleDerivationStream {
     Callable callable;
     int currIndex = 0;
+    List<String> strings = new ArrayList<String>();
 
     public JsonValueStream(Example ex, Callable c) {
       callable = c;
+      String string = callable.childStringValue(0);
+      strings.add(string);
+      if (string.matches(".*[s?!.,]"))
```

Comment:
shows the influence of cannonicalUtterances. perhaps best handled at the tokenizer?